code/game/machinery/computer/warrant.dm

```diff
@@ -0,0 +1,179 @@
+			<tr>
+			<th>Crime</th>
+			<th>Details</th>
+			<th>Author</th>
+			<th>Time Added</th>
+			</tr>"}
+			for(var/datum/data/crime/c in current.fields["mi_crim"])
+				dat += {"<tr><td>[c.crimeName]</td>
+				<td>[c.crimeDetails]</td>
+				<td>[c.author]</td>
+				<td>[c.time]</td>
+				</tr>"}
+			dat += "</table>"
+
+			dat += "<br>Major Crimes:"
+			dat +={"<table style="text-align:center;" border="1" cellspacing="0" width="100%">
+			<tr>
+			<th>Crime</th>
+			<th>Details</th>
+			<th>Author</th>
+			<th>Time Added</th>
+			</tr>"}
+			for(var/datum/data/crime/c in current.fields["ma_crim"])
+				dat += {"<tr><td>[c.crimeName]</td>
+				<td>[c.crimeDetails]</td>
+				<td>[c.author]</td>
+				<td>[c.time]</td>
+				</tr>"}
+			dat += "</table>"
+		else
+			dat += {"<span>** No security record found for this ID **</span>"}
+	else
+		dat += {"<a href='?src=[REF(src)];choice=Login'>------------</a><hr>"}
+
+	var/datum/browser/popup = new(user, "warrant", "Security Warrant Console", 600, 400)
+	popup.set_content(dat.Join())
+	popup.set_title_image(user.browse_rsc_icon(src.icon, src.icon_state))
+	popup.open()
+
+/obj/machinery/computer/warrant/Topic(href, href_list)
+	if(..())
+		return
+
+	switch(href_list["choice"])
+		if("Login")
+			var/mob/M = usr
+			scan = M.get_idcard(TRUE)
+			if(scan && istype(scan))
+				for(var/datum/data/record/R in GLOB.data_core.security)
+					if(R.fields["name"] == scan.registered_name)
+						current = R
+				playsound(src, 'sound/machines/terminal_on.ogg', 50, 0)
+		if("Logout")
+			current = null
+			scan = null
+			playsound(src, 'sound/machines/terminal_off.ogg', 50, 0)
+
+		if("Pay")
+			for(var/datum/data/crime/p in current.fields["citation"])
+				if(p.dataId == text2num(href_list["cdataid"]))
+					var/obj/item/holochip/C = usr.is_holding_item_of_type(/obj/item/holochip)
+					if(C && istype(C))
+						var/pay = C.get_item_credit_value()
+						if(!pay)
+							to_chat(usr, "<span class='warning'>[C] doesn't seem to be worth anything!</span>")
+						else
+							var/diff = p.fine - p.paid
+							GLOB.data_core.payCitation(current.fields["id"], text2num(href_list["cdataid"]), pay)
+							to_chat(usr, "<span class='notice'>You have paid [pay] credit\s towards your fine</span>")
+							if (pay == diff || pay > diff || pay >= diff)
+								investigate_log("Citation Paid off: <strong>[p.crimeName]</strong> Fine: [p.fine] | Paid off by [key_name(usr)]", INVESTIGATE_RECORDS)
+								to_chat(usr, "<span class='notice'>The fine has been paid in full</span>")
+							qdel(C)
+							playsound(src, "terminal_type", 25, 0)
+					else
+						to_chat(usr, "<span class='warning'>Fines can only be paid with holochips</span>")
+	updateUsrDialog()
+	add_fingerprint(usr)
+
+
+/obj/machinery/computer/warrant/emp_act(severity)
+	. = ..()
+
+	if(stat & (BROKEN|NOPOWER) || . & EMP_PROTECT_SELF)
+		return
+
+/obj/machinery/computer/warrant/AltClick(mob/user)
```

Comment:
This and `eject_id` are now unused since you don't grab the id anymore?

Reply:
its ded